src/title.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import {Text1, Text2, Text5} from './text';
+import {useTheme} from './hooks';
+import Inline from './inline';
+import {createUseStyles} from './jss';
+import {pxToRem} from './utils/css';
+
+const useStyles = createUseStyles((theme) => ({
+    right: {
+        display: 'flex',
+        alignItems: 'center',
+        paddingLeft: 16,
+        height: ({desktopLinkHeight}) => pxToRem(desktopLinkHeight),
+        [theme.mq.tabletOrSmaller]: {
+            height: ({mobileLinkHeight}) => pxToRem(mobileLinkHeight),
+        },
+    },
+}));
+
+type TitleLayoutProps = {
+    title: React.ReactElement;
+    desktopLinkHeight: number;
+    mobileLinkHeight: number;
+    right?: React.ReactNode;
+};
+
+const TitleLayout = ({
+    title,
+    right,
+    desktopLinkHeight,
+    mobileLinkHeight,
+}: TitleLayoutProps): React.ReactElement => {
+    const classes = useStyles({desktopLinkHeight, mobileLinkHeight});
+
+    if (!right) {
+        return title;
+    }
+
+    return (
+        <Inline space="between">
+            {title}
+            <div className={classes.right}>
+                <Text2 regular>{right}</Text2>
+            </div>
+        </Inline>
+    );
+};
+
+type TitleProps = {
+    children: React.ReactNode;
+    id?: string;
+    right?: React.ReactNode;
+    as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+};
+
+export const Title1 = ({children, as = 'h3', id, right}: TitleProps): React.ReactElement => {
+    const theme = useTheme();
+
+    return (
+        <TitleLayout
+            desktopLinkHeight={20}
+            mobileLinkHeight={16}
```

Comment:
these magic values are the line-height of `Text1` in desktop/mobile, I didn't find an automatic way of doing this without having to repeat these values here.